upstream_sync.py

```diff
@@ -24,10 +24,23 @@
 tmp_dir = '/var/tmp/upstream_sync-{0}'.format(user)
 
 
+def make_dir(dir_path, mode=0700):
+    """ checks if a directory exists and create it if necessary
+    :param mode: mode for makedirs
+    :param dir_path: dir path for check
+
+    :return: None
+    """
+    if not os.path.isdir(dir_path):
+        try:
+            os.makedirs(dir_path, mode)
+        except IOError as e:
+            print e
```

Comment:
We  eed to do a bit more than just print the error. If we can't create the directory there is no expectation the sync and createrepo commands will be successful. The quick solution is to print the error and then exit. The more elegant solution may be to handle the error and continue to the next repository.